### PR TITLE
boards: arm: Add support for Actinius Icarus Bee and SoM (nRF9160)

### PR DESCRIPTION
**Pull request by alextsam**
This PR adds support for 2 new boards based on the nRF9160.

These two board definitions are largely similar and they are derived from the boards/arm/actinius_icarus board

Signed-off-by: Alex Tsamakos <alex@actinius.com>

### CODE REVIEW

**Comment by alextsam** [comment]
@MaureenHelm @galak @nordic-krch @jakub-uC @nashif Could you please let me know if there is something I could do to get this merged soon? Thanks a lot in advance!

**Comment by MaureenHelm** [comment]
> @MaureenHelm @galak @nordic-krch @jakub-uC @nashif Could you please let me know if there is something I could do to get this merged soon? Thanks a lot in advance!

@alextsam, the merge window is currently closed to new features until the v2.5.0 release is complete, which is scheduled for 12 Feb. We'll have some more bandwidth to review this after that.

**Comment by alextsam** [comment]
> > @MaureenHelm @galak @nordic-krch @jakub-uC @nashif Could you please let me know if there is something I could do to get this merged soon? Thanks a lot in advance!
> 
> @alextsam, the merge window is currently closed to new features until the v2.5.0 release is complete, which is scheduled for 12 Feb. We'll have some more bandwidth to review this after that.

@MaureenHelm Ah I was not aware of the feature window, thank you so much for the information!

**Comment by MaureenHelm** [comment]
> @MaureenHelm Ah I was not aware of the feature window, thank you so much for the information!

This gets announced on the [mailing lists](https://lists.zephyrproject.org/g/main/subgroups). I recommend subscribing to devel or announcements to receive future updates.

**Review by MaureenHelm — APPROVED** [review]

**Comment by alextsam** [comment]
Could you please remove the stale label? Thank you!

**Comment by ioannisg** [comment]
will take a look

**Review by ioannisg — APPROVED** [review]
Fine with me. @alextsam you might want to align this with the latest version of nrf1960dk_nrf9160 board definition, but that's not mandatory.

**Comment by ioannisg** [comment]
I would also like @anangl to take a quick look.

**Comment by alextsam** [comment]
> Fine with me. @alextsam you might want to align this with the latest version of nrf1960dk_nrf9160 board definition, but that's not mandatory.

Please DNM just yet, I will align with the latest changes of the nrf9160dk_nrf9160 today 
Thank you @ioannisg

File: boards/arm/actinius_icarus_bee/actinius_icarus_bee_common.dts [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+	status = "okay";
+};
+
+&pwm2 {
+	status = "okay";
+};
+
+&pwm3 {
+	status = "okay";
+};
```
Comment (anangl) [inline boards/arm/actinius_icarus_bee/actinius_icarus_bee_common.dts:163]: Why enable these HW instances without assigning any pins to them? The PWM driver will consume (when enabled in Kconfig) some RAM and flash for data structures related to those (not much but still), but they will not be very useful (`pwm_pin_set_*` functions will fail for them, regardless of the supplied `pwm` parameter).

**Review by ioannisg — CHANGES_REQUESTED** [review]
nacking temporarily, until the changes are applied

**Comment by galak** [comment]
@alextsam any updates here.  Code freeze for getting this in is on Friday May 7th.

**Review by anangl — APPROVED** [review]

**Comment by alextsam** [comment]
Thank you for your patience, should be good to go now!

**Review by ioannisg — APPROVED** [review]

**Pull request merged**